src/modules/uORB/uORBManager.cpp

```diff
@@ -354,32 +354,27 @@ int uORB::Manager::node_open(const struct orb_metadata *meta, bool advertiser, i
 		*instance = 0;
 	}
 
-	/* we may need to advertise the node... */
-	if (fd < 0) {
+	/* Is a publisher or a new node */
+	if (advertiser || fd < 0) {
 
 		ret = PX4_ERROR;
 
 		if (get_device_master()) {
 			ret = _device_master->advertise(meta, advertiser, instance);
 		}
+	}
 
-		/* it's OK if it already exists */
-		if ((ret != PX4_OK) && (EEXIST == errno)) {
```

Comment:
Now don't need to judge this condition. We judge in DeviceMaster. If the node does not exist, then the node must not be registered to the file system.